exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/indices/ProofListIndexProxy.java

```diff
@@ -173,34 +172,32 @@ private ProofListIndexProxy(NativeHandle nativeHandle, IndexAddress address, Vie
   }
 
   /**
-   * Returns a proof that an element exists at the specified index in this list.
+   * Returns a proof of either existence or absence of an element at the specified index
+   * in this list.
    *
    * @param index the element index
    * @throws IndexOutOfBoundsException if the index is invalid
    * @throws IllegalStateException if this list is not valid
    */
   public UncheckedListProof getProof(long index) {
-    checkElementIndex(index, size());
-
     ListProof listProof = nativeGetProof(getNativeHandle(), index);
     return new UncheckedListProofAdapter<>(listProof, this.serializer);
   }
 
   private native ListProof nativeGetProof(long nativeHandle, long index);
 
   /**
-   * Returns a proof that some elements exist in the specified range in this list.
+   * Returns a proof of either existence or absence of some elements in the specified range
+   * in this list.
```

Comment:
If some elements are present in the list, but some — are not (i.e., the requested range exceeds its size [or crosses the list boundary]), a proof of absence is returned.